### PR TITLE
do not add Town06 and Town07 by default

### PR DESCRIPTION
**Pull request by Ruishenl**
@adscib  can you try this. Just add `-mc` flag . `python carla_demo.py --location=carla:Town10HD:3way_Stop -c 10 -l 60 -ep 2 -mc  --api_key= `

### CODE REVIEW

**Comment by AlirezaMorsali** [comment]
Not very important but we may consider renaming the docker file to `Dockerfile.additional-maps` to conform with Docker specifications.

**Comment by Ruishenl** [comment]
> Not very important but we may consider renaming the docker file to `Dockerfile.additional-maps` to conform to Docker specifications.

Done

File: examples/carla/docker-compose.yml [hunk 1]
```diff
@@ -2,9 +2,10 @@ version: "3.3"
 
 services:
   server:
-    build:
-      context: .
-      dockerfile: Dockerfile
+    image: "carlasim/carla"
```
Comment (AlirezaMorsali) [inline examples/carla/docker-compose.yml:5]: How about using `carlasim/carla:0.9.13` here too?

Comment (Ruishenl) [reply examples/carla/docker-compose.yml:5]: Sure, i was trying to save you guys some time to download another image

File: examples/carla/pygame_display.py [hunk 2]
```diff
@@ -9,11 +9,11 @@ def __init__(self, width, height):
...
-# Camera sensor callback, reshapes raw data from camera into 2D RGB and applies to PyGame surface
-def pygame_callback(data, obj):
-    data.convert(carla.ColorConverter.Raw)
-    img = np.reshape(np.copy(data.raw_data), (data.height, data.width, 4))
-    img = img[:, :, :3]
-    img = img[:, :, ::-1]
-    obj.surface = pygame.surfarray.make_surface(img.swapaxes(0, 1))
+    # Camera sensor callback, reshapes raw data from camera into 2D RGB and applies to PyGame surface
+    @staticmethod
+    def pygame_callback(data, obj):
```
Comment (adscib) [inline examples/carla/pygame_display.py:14]: If you flip the arguments, it's not a static method anymore...

File: examples/carla/docker-compose.yml [hunk 3]
```diff
@@ -2,9 +2,10 @@ version: "3.3"
 
 services:
   server:
-    build:
-      context: .
-      dockerfile: Dockerfile
+    image: "carlasim/carla:0.9.13"
+#    build:
+#      context: .
+#      dockerfile: Dockerfile-additional-maps
```
Comment (adscib) [inline examples/carla/docker-compose.yml:8]: Should there be some comment about uncommenting this to get additional maps?

**Review by adscib — CHANGES_REQUESTED** [review]

File: examples/carla/carla_simulator.py [hunk 4]
```diff
@@ -278,7 +278,7 @@ def __init__(
         image_h = camera_bp.get_attribute("image_size_y").as_int()
         # Instantiate objects for rendering and vehicle control
         self.renderObject = RenderObject(image_w, image_h)
-        self.camera.listen(lambda image: pygame_callback(image, self.renderObject))
+        self.camera.listen(lambda image: self.renderObject.pygame_callback(image, self.renderObject))
```
Comment (adscib) [inline examples/carla/carla_simulator.py:281]: You also need to adjust this, which conveniently doesn't need a lambda anymore.

File: examples/carla/docker-compose.yml (hunk 3 above)
Comment (adscib) [reply examples/carla/docker-compose.yml:9]: That's good, but also add a sentence to README to point this out.

**Review by adscib — CHANGES_REQUESTED** [review]

**Review by adscib — APPROVED** [review]
I haven't checked that this still runs, so please do so before merging.

**Comment by Ruishenl** [comment]
Checked and it still runs

**Pull request merged**